### PR TITLE
fix email receipt flag for recurring record

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Reviewer's commit of https://github.com/civicrm/civicrm-core/pull/19210

### CODE REVIEW

**Comment by yashodha** [comment]
@eileenmcnaughton

are the failing tests related?

**Comment by eileenmcnaughton** [comment]
@yashodha the 2 membership tests were a once-a-year-bug in the tests I think (also affect the other runs). the other is in the new test but didn't fail locally & could also be a new-years's bug. I've pushed up a cleanup patch which will trigger the tests to run again so we can see

**Comment by demeritcowboy** [comment]
jenkins test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by sunilpawar** [comment]
@eileenmcnaughton 
> CRM_Contribute_BAO_ContributionRecurTest::testContributionEmailReceipt
> ERROR USERINFO: INSERT INTO `civicrm_contribution_recur` (`contact_id` , `amount` , `currency` , `frequency_unit` , `frequency_interval` , `installments` , `start_date` , `create_date` , `modified_date` , `end_date` , `processor_id` , `contribution_status_id` , `is_test` , `cycle_day` , `next_sched_contribution_date` , `failure_count` , `auto_renew` , `payment_processor_id` , `financial_type_id` , `payment_instrument_id` ) VALUES ( 46 ,  3 , 'USD' , 'week' ,  1 ,  2 ,  0 ,  0 ,  0 ,  0 , '643411460836' ,  1 ,  0 ,  1 ,  0 ,  0 ,  0 ,  1 ,  1 ,  1 )  [nativecode=1292 ** Incorrect datetime value: '0' for column 'start_date' at row 1]

**Comment by eileenmcnaughton** [comment]
@sunilpawar does it fail there locally for you? It didn't for me yesterday - I'm wondering if there is a mysql config in play (@demeritcowboy does this seem familiar?)

File: tests/phpunit/CRM/Contribute/BAO/ContributionRecurTest.php [hunk 1]
```diff
@@ -564,4 +565,32 @@ public function yearMonthProvider() {
...
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public function testContributionEmailReceipt(): void {
+    $createParams = $this->_params;
+    unset($createParams['trxn_id'], $createParams['invoice_id']);
+
+    // pass null value to is_email_receipt
+    $createParams['is_email_receipt'] = NULL;
+    $recurring1 = CRM_Contribute_BAO_ContributionRecur::add($createParams);
```
Comment (demeritcowboy) [inline tests/phpunit/CRM/Contribute/BAO/ContributionRecurTest.php:579]: I think the test failure is just that the add() function doesn't take this type of params. The params format seems intended for the api which converts relative dates like 'yesterday' into actual dates. If you give that to add() then it tries to literally insert 'yesterday' into a date field.

Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Contribute/BAO/ContributionRecurTest.php]: Ok - I fixed it up to call the api - hopefully it passes now

**Pull request merged**